modulegen/dependabot.go

```diff
@@ -1,96 +1,20 @@
 package main
```

Comment:
I'd try to move all dependabot code from the main package to the new `pkg/dependabot`, so that the main.go file just consumes what the `dependabot` package exposes.

On the other hand, let's think if we want to expose a public API for handling dependabot files, using the publicly available `pkg/dependabot` package (which could be consumed in the wild), or on the contrary keep it internal (i.e. `./internal/dependabot`).

Before moving on with this PR and #1504 and #1505, let's discuss first about the APIs that we want to create, and the project layout for that. Let's iterate in one single command, and once we are happy with it, let's move to the others. I'm open to have a slack chat/huddle or zoom call for sync if needed

Does it sound good to you?